scheduler/src/main/java/io/camunda/zeebe/scheduler/ActorJob.java

```diff
@@ -58,7 +57,7 @@ void execute(final ActorThread runner) {
       actorThread = null;
 
       // in any case, success or exception, decide if the job should be resubmitted
-      if (isTriggeredBySubscription() || (isAutoCompleting && runnable == null) || isDoneCalled) {
+      if (isTriggeredBySubscription() || (isAutoCompleting && runnable == null)) {
```

Comment:
:wrench:

`isAutoCompleting` is always true, except for dispatcher consumers for which `isTriggeredBySubscription` is true. That means we can get rid of `isAutoCompleting` too :tada:

Additionally, I'd suggest flipping the condition here so that we check `runnable == null` first to short-circuit for the majority of actor jobs.